### Problem Statement 1: 
 
Blood glucose levels for obese patients have a mean of 100 with a standard deviation of  15. A researcher thinks that a diet high in raw cornstarch will have a positive effect on blood glucose levels. A sample of 36 patients who have tried the raw cornstarch diet have a mean glucose level of 108. Test the hypothesis that the raw cornstarch had an effect or not. 

In [1]:
import math
pMean = 100 # populatin mean
pStd = 15 # population standard deviation
size_Sample = 36 # sample size
sMean = 108 # sample mean

# Hypothesis: 
# H0: 𝜇 = 100, # null hypothesis , no effect
# H1: 𝜇 ≠ 100, # alternate hypothesis, yes have effect
# This will be a two-tailed test

SE = (pStd)/math.sqrt(size_Sample) # Standard error
    
def Z_test(sMean, pMean, SE):  # Z test calculation
    Z_val = (sMean - pMean) / SE
    return Z_val

val = Z_test(sMean, pMean, SE)
print('The computed value of Z is: ', val)

The computed value of Z is:  3.2


In [2]:
import scipy
from scipy import stats
from scipy.stats import norm

# Now we need to find out the critical value for significance level 5% = 0.05 let say
# So, Alpha = 0.05
# As two-tailed test then, Alpha/2 = 0.025
# probability = 1 - alpha/2 = 1-0.025 = 0.975
# Gaussian Percent Point Function (ppf)
# define probability
p = 0.975
# retrieve value <= probability
value = norm.ppf(p) # returns observaton values for provided probability which is less than or equal to the provided probability
print('The critical value for Z is: ', value)
# confirm with cdf
p = norm.cdf(value) # Returns the probability for an observation equal to or lesser than a specific value from the distribution
print(p)

The critical value for Z is:  1.959963984540054
0.975


<font color='blue'>
    
The criticial value of Z is: Z = +1.96 and -1.96 at 5% level of significance for two-tailed test. <br>
Our computed value of Z is: Z = 3.2. <br>

So, computed value falls in rejection region. Hence we reject the Null Hypothesis. <br>
Hence the raw cornstarch had an effect on blood glucose levels for obese patients.

    
</font>

### Problem Statement 2: 
 
In one state, 52% of the voters are Republicans, and 48% are Democrats. In a second state, 47% of the voters are Republicans, and 53% are Democrats. Suppose a simple random sample of 100 voters are surveyed from each state. 
 
What is the probability that the survey will show a greater percentage of Republican voters in the second state than in the first state? 

In [3]:
P1 = 0.52 # 1st state republic voter
P2 = 0.47 # 2nd state republic voter

n1 = 100 # 1st state number of sampled voter
n2 = 100 # 2nd state number of sampled voter

q1 = 0.52 # 1st state republice voter in the sample
q2 = 0.47 # 2nd state republice voter in the sample

# meanDifference_sampleProportion = q1 - q2 = P1 - P2 = 0.52 - 0.47 = 0.05

𝜇 = 0.05 # sample mean


In [4]:
import math
# Calculate standard deviation
def std_dev(P1, n1, P2, n2):
    𝜎 = math.sqrt(((P1/n1)*(1-P1)) + (P2/n2)*(1-P2))
    return 𝜎
stdVal = std_dev(P1, n1, P2, n2)
print('Calculated standard deviation is: ', stdVal)

Calculated standard deviation is:  0.07061869440877536


In [5]:
# In this problem we need to find out the probability that q1 is less than q2.
# This is euqavalent to find out the probability that q1-q2 is less than zero.
# So, we need to find out Z-score for the random variable q1-q2 to find out the required probaibility.

# finding out Z-score
x = 0
𝜇 = 0.05
𝜎 = stdVal
z_score = (x-𝜇) / 𝜎

print('Calculated Z-score is: ', z_score)


Calculated Z-score is:  -0.7080278164104207


In [6]:
# Now will find out the probability of a z-score being -0.708 or less.
# P(Z <= 0.708)

z = 0.708
value1 = 1 - norm.cdf(z) # returns %value of the test takers below the specified number
print('So, the probability that the survey will show a greater percentage of Republican voters in the 2nd state than in the 1st state is: ', value1)


So, the probability that the survey will show a greater percentage of Republican voters in the 2nd state than in the 1st state is:  0.23947262873987984


<font color='blue'>

### Conclusions: <br>
So, the probability that the survey will show a greater percentage of Republican voters in the 2nd state than in the 1st state is:  0.24

</font>

### Problem Statement 3: 
 
You take the SAT and score 1100. The mean score for the SAT is 1026 and the standard deviation is 209. How well did you score on the test compared to the average test taker? 

In [7]:
x = 1100 # data point
𝜇 = 1026 # mean
𝜎 = 209 # standard deviation

def Z_score(x, 𝜇, 𝜎):
    
    z = (x - 𝜇)/ 𝜎
    
    return z

zScore = Z_score(x, 𝜇, 𝜎)

print('So, Z Score {} represents my score was {} standard deviation above the mean'.format(zScore, zScore) )

So, Z Score 0.35406698564593303 represents my score was 0.35406698564593303 standard deviation above the mean


In [8]:
import scipy
from scipy import stats
from scipy.stats import norm

# Now will check what percentange of test takers scored below me
# P(X<1100) = P(z<0.354)

z = 0.354
value = norm.cdf(z) # returns %value of the test takers below the specified number
print('So, percentage of the test takers scored below me: ', value*100)



So, percentage of the test takers scored below me:  63.83305583821042


In [9]:
# Now will check what percentange of test takers scored above me
# P(X>1100) = P(z>0.354) = 1 - P(z<0.354)

z = 0.354
value1 = 1 - norm.cdf(z) # returns %value of the test takers below the specified number
print('So, percentage of the test takers scored below me: ', value1*100)


So, percentage of the test takers scored below me:  36.16694416178958


<font color='blue'>

### Conclusions: <br>
My score was 0.354 standard deviaton above the mean. <br>
63.83% of the test takers scored below me. <br>
36.17% of the test takers scored above me. <br>

</font>